### ISSUE TITLE
Ques: Is there a reason to force the operationId to camelcase

### ISSUE THREAD

**Issue by abhishekdev** [oazapfts/oazapfts#591]
### Question

Is there a reason to force the operationID to camelcase? 

**Current Behavior**
e.g.  A operation ID like `EvaluateUserIAM` will get generated as `evaluateUserIam`

**Expected behavior**
Based on the spec guide (reffered below) I guess if an operation ID is found on the schema it should not be mutated.

The [openID spec](https://swagger.io/specification/#fixed-fields-8) mentions that the operation ID is **case sensitive**, however since the generator is mutating the operation name to be camelcase it causes confusion for consuming clients.

<img width="891" alt="Screenshot 2024-02-28 at 5 33 09 PM" src="https://github.com/oazapfts/oazapfts/assets/227926/68a1d790-b4ad-4e13-bdb2-ddbe517465fb">

**Source Reference**
https://github.com/oazapfts/oazapfts/blob/702c4008d53280f46f9827df2f04eef1ce3690c7/packages/codegen/src/generate.ts#L120-L125

**Comment by Xiphe** [comment]
I tend to say: Probably no specific reason.

For me this makes sense though as Commonly in javascript/typescript functions and methods start with a lower case and only classes and types use uppercase.

**Comment by abhishekdev** [comment]
Thanks for sharing your thoughts @Xiphe.  

It seems there is already a check and graceful fallback built into the logic to check for invalid JS/TS characters (i.e. regex + `isValidIdentifier` check). Given this checks are in place, should common JS code convention override the spec at this code generation layer though? 

As long as valid code is generated, I guess the API naming spec (=> source of truth) should be given precedence which might be using its own contextual naming convention. 

e.g. Even if the the API name is `evaluateUserIAM` to start with, where `IAM` is an abbreviation), the current naming would force it to `evaluateUserIam` creating a cognitive load for the users consuming the same API over different platforms (via different code generators)

**Comment by Xiphe** [comment]
I get your point and principally agree.

Please bare in mind that when we change this, it will potentially have wide-ranging impacts. It would potentially cause a lot of breaking changes for the clients generated. I'd therefore move this behind a feature flag to opt-into rather then change default behavior.

With this in mind, personally I find the additional cognitive load negligible as such I'm not sure If the additional complexity on this system is worth it. 

I'd still be open to learn if others would also be interested in having this behavior. Might also be a good case for the [Plugin System](https://github.com/oazapfts/oazapfts/issues/562) that @fimion is currently championing. (For this se should maybe add a naming related hook)

**Comment by abhishekdev** [comment]
I agree! Changing this should definitely considered as a breaking change. 

The ability to opt-in is a reasonable path forward. 

The concept of the "plugin system" sounds very interesting. I'll take a look over the weekend and If I can contribute.
